hummingbot/strategy/pure_market_making/pure_market_making.pyx

```diff
@@ -82,10 +84,13 @@ cdef class PureMarketMakingStrategy(StrategyBase):
                     split_order_levels_enabled: bool = False,
                     bid_order_level_spreads: List[Decimal] = None,
                     ask_order_level_spreads: List[Decimal] = None,
-                    should_wait_order_cancel_confirmation = True,
+                    should_wait_order_cancel_confirmation: bool = True,
+                    moving_price_band: Optional[MovingPriceBand] = None
                     ):
         if order_override is None:
             order_override = {}
+        if moving_price_band is None:
+            moving_price_band = DisabledMovingPriceBand()
```

Comment:
@Protechris personally I don't really like this method you're using to disable this functionality using the "DisabledMovingPriceBand" class, seems to add unnecessary cycles rather than just disabling it on init if `None`, it's also not the way things are usually implemented when disabled.

Reply:
Hello @TheHolyRoger. I asked him to do that change (you can see that in the previous comments). He is modeling the different moving price bands strategies using objects, and then achieves the result of enabling/disabling the functionality with polymorphism instead of using if conditions. In my opinion that makes the code easier to maintain and extend in the future

Reply:
Hi @rapcmia, sorry for the late reply, I am a little preoccupied lately. Give me a few days and let me see if I can come up with a note the log for this. thanks.

Hi @TheHolyRoger and @aarmoa, thanks for the feedback and clarification. much appreciated. I will take note on this for future PR.

Reply:
@Protechris we had a meeting with the technical DAO members to decide which approach to take, and the decision was that we should come back to your initial implementation, where the `DisabledMovingPriceBand` does not exist and a boolean flag is used to enable/disable the moving price. Would it be possible for you to undo the change?

Reply:
Yeah. I have revert it back. thanks.